linode/manager
# feat: [M3-5915] Accessible graph data

## Description 📝
Apart from a general description as to what the chart is representing, there's is currently no information details available to screen readers users. This is an accessibility issue described in M3-5915.

This PR attempts to improve this limitation by adding a visually hidden data table to each line chart. Please refer to the self review for implementation details as well as the code inline comments.

This PR follows the [accessibility guidelines for the canvas element](https://pauljadam.com/demos/canvas.html).

Note: this only targets `<Linegraphs />` and does not address Longview graphs (I added a separate M3-6517 ticket for those, which is lower priority).

## Preview 📷
There is no visual change associated with this PR

## How to test 🧪
A few ways this PR should be tested.

1. Use an accessibility tool to inspect the new data tables for various graphs
2. Inspect the tables visually (locally) by removing the `sx={visuallyHidden}` prop at https://github.com/abailly-akamai/manager/blob/d09521ff8b924106c05a29df8a19d962f68c35a9/packages/manager/src/components/LineGraph/AccessibleGraphData.tsx#L77 - make sure the filters are working for the tabled data as well.
3. Use the PR preview link to inspect that page performance isn't affected by much, and that graphs that rely on pagination also get their data tables re-rendered.  (ex: /linodes/[id]/networking)
